oauth2/oauth2-jose/src/test/java/org/springframework/security/oauth2/jwt/NimbusJwtDecoderTests.java

```diff
@@ -678,6 +760,22 @@ public void decodeWhenCacheIsConfiguredAndValueLoaderErrorsThenThrowsJwtExceptio
 		// @formatter:on
 	}
 
+	// gh-11621
+	@Test
+	public void decodeWhenCacheIsConfiguredAndParseFailsOnCachedValueThenThrowsJwtException() {
+		Cache cache = new ConcurrentMapCache("test-jwk-set-cache");
+		cache.put(JWK_SET_URI, "{}");
```

Comment:
I wonder why the code would add a malformed JWK Set to the cache in the first place. Is there a way to instead ensure that what goes into the cache is well-formed?

Reply:
Good question. This is one of those code-coverage for its own sake tests. Though technically it could happen outside of this component, I don't believe it's possible to happen normally. Shall I just remove the test?

Reply:
I think the test that's needed is to ensure that what goes into the cache is well-formed.